packages/app-mobile/components/NoteBodyViewer/hooks/useOnMessage.ts

```diff
@@ -1,7 +1,25 @@
 import { useCallback } from 'react';
 const shared = require('@joplin/lib/components/shared/note-screen-shared');
 
-export default function useOnMessage(onCheckboxChange: Function, noteBody: string, onMarkForDownload: Function, onJoplinLinkClick: Function, onResourceLongPress: Function) {
+export type HandleMessageCallback = (message: string)=> void;
+export type OnMarkForDownloadCallback = (resource: { resourceId: string })=> void;
+
+interface MessageCallbacks {
+	onMarkForDownload?: OnMarkForDownloadCallback;
+	onJoplinLinkClick: HandleMessageCallback;
+	onResourceLongPress: HandleMessageCallback;
+	onRequestEditResource?: HandleMessageCallback;
+	onCheckboxChange: HandleMessageCallback;
+}
+
+export default function useOnMessage(
+	noteBody: string,
+	callbacks: MessageCallbacks
+) {
+	const {
+		onMarkForDownload, onResourceLongPress, onCheckboxChange, onRequestEditResource, onJoplinLinkClick,
+	} = callbacks;
```

Comment:
`callbacks` is destructured into component variables here to prevent the dependency hook from requesting that the `useCallback` depend on `callbacks` (which, because `({ a: 1 }) !== ({ a: 1 })`, would cause the `useCallback`'s dependencies to change every time `useOnMessage` is called).

To prevent this change from being undone, I should add a comment here.